src/wwwroot/services/auth.js

```diff
@@ -136,12 +164,14 @@
         logOut();
       } else if (token != encodedToken) {
         encodedToken = token;
-        decodedToken = jwtHelper.decodeToken(encodedToken);
+        decodeToken();
       }
     }
 
+
     function logOut() {
       decodedToken = null;
+      permissons = null;
```

Comment:
I think think that permissons object could be considered as part of _decodedToken_ but it is a valid approach to have an independent variable for it.